test.py

```diff
@@ -0,0 +1,37 @@
+"""Unit tests for main module."""
+
+import unittest
+from unittest import mock
+from fastapi.testclient import TestClient
+
+import main
+
+
+class RoutingTest(unittest.TestCase):
+    """Unit tests for routing and each operation."""
+
+    def setUp(self):
+        patcher_load_config_file = mock.patch("main.load_config_file")
+        patcher_get_client = mock.patch("main.get_client")
+        self.mocked_load_config_file = patcher_load_config_file.start()
+        self.mocked_get_client = patcher_get_client.start()
+        self.addCleanup(patcher_load_config_file.stop)
+        self.addCleanup(patcher_get_client.stop)
+
+    @mock.patch.dict("main.configs", {"master_path": "master_path/"})
+    def test_list_directory(self):
+        test_list = ["dir1/", "dir2/", "file1.py", "file2.py"]
+        self.mocked_get_client.return_value.list_directory.return_value = test_list
```

Comment:
Defining `mocked_client = self.mocked_get_client.return_value` could be helpful.